File: app/src/main/java/com/suzei/racoon/ui/auth/signup/SignUpContract.java
package com.suzei.racoon.ui.auth.signup;

import com.suzei.racoon.ui.base.Contract;

public interface SignUpContract {

    interface View extends Contract.ProgressView {

        void setPresenter(Presenter presenter);

        void onRegisterSuccess();

        void onRegisterFailure(String message);

    }


    interface Presenter {

        void createAccount(String email, String password, String name);

    }

}

File: app/src/main/java/com/suzei/racoon/ui/auth/signup/SignUpContract.kt
package com.suzei.racoon.ui.auth.signup

import com.suzei.racoon.ui.base.Contract.ProgressView

interface SignUpContract {
    interface View : ProgressView {
        fun setPresenter(presenter: Presenter?)
        fun onRegisterSuccess()
        fun onRegisterFailure(message: String?)
    }

    interface Presenter {
        fun createAccount(email: String?, password: String?, name: String?)
    }
}